src/NexusMods.App.UI/Pages/Downloads/ViewModels/DownloadTaskViewModel.cs

```diff
@@ -1,129 +1,83 @@
-    }
-
-    public string Name => _task.FriendlyName;
-
-    public string Version
-    {
-        get
+        
+        this.WhenActivated(d =>
         {
-             if (_task is IHaveDownloadVersion version)
-                 return version.Version;
-
-             return Language.DownloadTaskViewModel_Field_Unknown;
-        }
+            _task.WhenAnyValue(t => t.PersistentState.FriendlyName)
+                .OnUI()
+                .Select(t => t)
+                .BindTo(this, x => x.Name)
+                .DisposeWith(d);
+            
+            _task.WhenAnyValue(t => t.PersistentState.Version)
+                .OnUI()
+                .BindTo(this, x => x.Version)
+                .DisposeWith(d);
+
+            _task.WhenAnyValue(t => t.PersistentState.Status)
+                .OnUI()
+                .BindTo(this, x => x.Status)
+                .DisposeWith(d);
+
+            _task.WhenAnyValue(t => t.Downloaded)
+                .OnUI()
+                .Select(s => s.Value)
+                .BindTo(this, x => x.DownloadedBytes)
+                .DisposeWith(d);
+
+            _task.WhenAnyValue(t => t.PersistentState.Size)
+                .OnUI()
+                .Select(s => s.Value)
+                .BindTo(this, x => x.SizeBytes)
+                .DisposeWith(d);
+            
+            _task.WhenAnyValue(t => t.PersistentState.GameDomain)
+                .OnUI()
+                .Select(g => g.ToString())
+                .BindTo(this, x => x.Game)
+                .DisposeWith(d);
+
+            _task.WhenAnyValue(t => t.Bandwidth)
+                .OnUI()
+                .Select(b => b.Value)
+                .BindTo(this, x => x.Throughput)
+                .DisposeWith(d);
+
+        });
     }
 
-    public string Game
-    {
-        get
-        {
-            if (_task is IHaveGameName name)
-                return name.GameName;
-
-            return Language.DownloadTaskViewModel_Field_Unknown;
-        }
-    }
+    [Reactive] public string Name { get; set; } = "";
 
-    public DownloadTaskStatus Status => _task.Status;
+    [Reactive] public string Version { get; set; } = "";
+    [Reactive] public string Game { get; set; } = "";
+    
+    public EntityId TaskId => _task.PersistentState.Id;
 
-    public long DownloadedBytes => _task.DownloadedSizeBytes;
+    [Reactive] public DownloadTaskStatus Status { get; set; } 
 
-    public long SizeBytes
-    {
-        get
-        {
-            if (_task is IHaveFileSize size)
-                return size.SizeBytes;
+    [Reactive] public long DownloadedBytes { get; set; }
 
-            return 0;
-        }
-    }
+    [Reactive] public long SizeBytes { get; set; }
 
-    public long Throughput => _task.CalculateThroughput();
+    [Reactive] public long Throughput { get; set; }
 
     public void Cancel() => _task.Cancel();
     public void Suspend() => _task.Suspend();
     public void Resume() => _task.Resume();
```

Comment:
I think these should probably be async tasks, there is some weird stuff going on with async and non async methods here.
I missed this when redid some of this code